pkg/server/server.go

```diff
@@ -370,44 +371,51 @@ func (s *ProxyServer) Proxy(stream client.ProxyService_ProxyServer) error {
 	recvCh := make(chan *client.Packet, xfrChannelSize)
 	stopCh := make(chan error)
 
-	go s.serveRecvFrontend(stream, recvCh)
+	labels := runpprof.Labels(
+		"serverID", s.serverID,
+		"serverCount", strconv.Itoa(s.serverCount),
+		"userAgent", strings.Join(userAgent, ", "),
+	)
+	go runpprof.Do(context.Background(), labels, func(context.Context) { s.serveRecvFrontend(stream, recvCh) })
 
 	defer func() {
 		klog.V(2).InfoS("Receive channel from frontend is stopping", "userAgent", userAgent, "serverID", s.serverID)
 		close(recvCh)
 	}()
```

Comment:
nit: move this to above the go-routine call, to group the 2 go routines together.

Reply:
The placement of the defered recvCh close has significant consequences and is not something I want conflate with this change.